src/smartpeak/include/SmartPeak/ui/Widget.h

```diff
@@ -85,14 +85,20 @@ namespace SmartPeak
   class GenericTableWidget : public Widget
   {
   public:
-    typedef void(SessionHandler::* DataGeterMethod)(const SequenceHandler& sequence_handler, SessionHandler::GenericTableData& generic_table_data);
+    typedef void(SessionHandler::* DataGetterMethod)(const SequenceHandler& sequence_handler, SessionHandler::GenericTableData& generic_table_data);
     typedef Eigen::Tensor<bool, 1>(SessionHandler::* DataFilterMethod)(const Eigen::Tensor<std::string, 2>& to_filter) const;
 
+    /*
+    * GenericTableWidget.
+    * 
+    * A data getter can be provided so that data will be retreive from SessionHandler automatically in the draw method.
```

Comment:
In my opinion, this is quite nice how you have maintained the separation of the logic with the data as well as the methods for getting the data and filtering it.  This should make things much easier down the road.